### PR TITLE
Simplify register generation code

### PR DESCRIPTION
**Pull request by therealprof**

### CODE REVIEW

**Comment by therealprof** [comment]
bors try

File: src/generate/register.rs [hunk 1]
```diff
@@ -223,113 +223,77 @@ pub fn fields(
...
+            evs,
             fields,
             parent,
             all_registers,
             peripheral,
             all_peripherals,
         )?;
 
-        let pc_r = &f.pc_r;
-        let mut pc_w = &f.pc_r;
```
Comment (burrbull) [inline src/generate/register.rs]: It's important!

Comment (therealprof) [reply src/generate/register.rs]: It's kind of hard to see without any comments in the code. ;) I'll have a look at the build results and try to incorporate it in my tests.

Comment (burrbull) [reply src/generate/register.rs]: Reader and writer must use one common `Enum_A` unless field have `enumeratedValues`: one for `read-only` and other for `write-only` (`Enum_AW`).

File: src/generate/register.rs [hunk 2]
```diff
@@ -459,29 +422,28 @@ pub fn fields(
...
             if let Some((evs, base)) = lookup_filter(&lookup_results, Usage::Write) {
                 let variants = Variant::from_enumerated_values(evs)?;
 
-                if variants.len() == 1 << f.width {
+                if variants.len() == 1 << width {
                     unsafety = None;
                 }
 
                 if Some(evs) != evs_r.as_ref() {
-                    pc_w = &f.pc_w;
```
Comment (burrbull) [inline src/generate/register.rs]: And this.

**Comment by therealprof** [comment]
Thanks for the explanation. I added a comment to that effect.

**Comment by therealprof** [comment]
bors try

**Pull request merged**